packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -40,65 +40,51 @@ export const CategoriesProvider = ({treeRoot = {}, children, locale}) => {
 
     const api = useCommerceAPI()
     const [root, setRoot] = useState({
-        // map over the server-provided cat
+        // Clone the server-provided category
         ...treeRoot[DEFAULT_ROOT_CATEGORY],
         [itemsKey]: treeRoot[DEFAULT_ROOT_CATEGORY]?.[itemsKey]?.map((item) => ({
             ...item
         }))
     })
 
     const fetchCategoryNode = async (id, levels = 1) => {
+        const storageKey = `${LOCAL_STORAGE_PREFIX}${id}-${locale}`
+        const storageItem = JSON.parse(window.localStorage.getItem(storageKey))
+
         // return early if there's one that is less than stale time
-        const storageItem = JSON.parse(
-            window.localStorage.getItem(`${LOCAL_STORAGE_PREFIX}${id}-${locale}`)
-        )
-        if (storageItem || Date.now() < storageItem?.fetchTime + CAT_MENU_STALE_TIME) {
+        if (storageItem && Date.now() < storageItem.fetchTime + CAT_MENU_STALE_TIME) {
             return storageItem
         }
+        // get and store fresh data for faster access / reduced server load
+        // TODO: Convert to useCategory hook when hooks are ready
         const res = await api.shopperProducts.getCategory({
             parameters: {
                 id,
                 levels
             }
         })
+        res.loaded = true
+        window.localStorage.setItem(
+            storageKey,
+            JSON.stringify({
+                ...res,
+                fetchTime: Date.now()
+            })
+        )
         return res
     }
 
-    useEffect(() => {
+    useEffect(async () => {
```

Comment:
Does `useEffect` behave differently when the callback returns `Promise<void>` instead of `void`? I don't think so, so I switched from promise chains to async/await. This is purely for readability, though, so can be reverted if there are unintended consequences.

Reply:
Did you test your change locally to check manually for expected behavior?

Reply:
Yeah, but this feels like the kind of thing that would have sneaky, hidden side effects, so I wanted to make sure.

Reply:
+1, we do not want to have `useEffect(async () => {`